www-servers/h2o/files/h2o.initd

```diff
@@ -0,0 +1,53 @@
+#!/sbin/openrc-run
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+extra_commands="configtest"
+extra_started_commands="reload"
+description="An optimized HTTP/1.x, HTTP/2 server"
+description_configtest="Run H2O' internal config check"
+description_reload="Reload the H2O configuration or upgrade the binary without losing connections"
+
+: ${config:="/etc/h2o/h2o.conf"}
+pidfile=`grep pid-file ${config} | cut -d' ' -f2`
+
+name="H2O"
+command="/usr/bin/h2o"
+command_args="-m daemon -c ${config}"
+required_files="$config"
+
+depend() {
+	use net
+	after logger
+}
+
+start_pre() {
+	if [ "${RC_CMD}" != "restart" ]; then
+		configtest || return 1
+	fi
+}
+
+stop_pre() {
+	if [ "${RC_CMD}" = "restart" ]; then
+		configtest || return 1
+	fi
+}
+
+reload() {
+	configtest || return 1
+	ebegin "Refreshing ${name} configuration"
+	kill -HUP `cat ${pidfile}` &>/dev/null
+	eend $? "Failed to reload ${name}"
+}
+
+configtest() {
+	ebegin "Checking ${name} configuration"
+
+	${command} -c ${config} -t &>/dev/null
+
+	if [ $? -ne 0 ]; then
+		${command} -c ${config} -t
```

Comment:
I decided to silence it because even on a successful test, the command spill some infos. Like this one:

```
[OCSP Stapling] testing for certificate file:/etc/ssl/foo.cert.pem
fetch-ocsp-response (using LibreSSL 2.5.4)
sending OCSP request to http://ocsp.int-x3.letsencrypt.org
/etc/ssl/foo.cert.pem: good
	This Update: Jun 29 10:00:00 2017 GMT
	Next Update: Jul  6 10:00:00 2017 GMT
verifying the response signature
verify OK (used: -VAfile /tmp/xzl_jaTbwa/issuer.crt)
[OCSP Stapling] stapling works for file:/etc/ssl/foo.cert.pem
configuration OK
```